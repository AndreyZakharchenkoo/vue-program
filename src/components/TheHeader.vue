<template>
  <header class="header">
    <div class="container">
      <the-logo/>
      <router-link to="/" class="header__icon">
        <img src="../assets/icon-search.svg" alt="Search icon">
      </router-link>
    </div>
  </header>
</template>

<script>
import TheLogo from '@/stories/TheLogo.vue'

export default {
  name: 'TheHeader',
  components: {
    TheLogo
  }
}
</script>

<style lang="scss" scoped>
.header {
  position: absolute;
  z-index: 100;
  top: 0;
  left: 0;
  right: 0;

  .container {
    display: flex;
    justify-content: space-between;
    align-items: center;
    min-height: 70px;
    padding-top: 20px;
    padding-bottom: 20px;
  }

  &__icon {
    width: 32px;
  }
}
</style>
