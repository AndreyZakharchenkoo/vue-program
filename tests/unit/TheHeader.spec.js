import { shallowMount } from '@vue/test-utils';
import TheHeader from '@/components/TheHeader.vue';

describe('TheHeader.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(TheHeader, {
      stubs: ['router-link', 'router-view'],
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('Renders correctly', () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
