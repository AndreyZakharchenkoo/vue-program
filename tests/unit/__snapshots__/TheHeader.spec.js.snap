// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHeader.vue Renders correctly 1`] = `
<header
  class="header"
>
  <div
    class="container"
  >
    <the-logo-stub />
     
    <router-link-stub
      class="header__icon"
      to="/"
    >
      <img
        alt="Search icon"
        src="../assets/icon-search.svg"
      />
    </router-link-stub>
  </div>
</header>
`;
